var/spack/repos/builtin/packages/lbann/package.py

```diff
@@ -107,20 +107,22 @@ class Lbann(CMakePackage):
     extends("python")
     depends_on('py-setuptools', type='build')
     depends_on('py-argparse', type='run', when='@:0.90,0.99: ^python@:2.6')
-    depends_on('py-configparser', type='run', when='@:0.90,0.99:')
-    depends_on('py-graphviz@0.10.1:', type='run', when='@:0.90,0.99:')
-    depends_on('py-matplotlib@3.0.0:', type='run', when='@:0.90,0.99:')
-    depends_on('py-numpy@1.16.0:', type=('build', 'run'), when='@:0.90,0.99:')
-    depends_on('py-onnx@1.3.0:', type='run', when='@:0.90,0.99:')
-    depends_on('py-pandas@0.24.1:', type='run', when='@:0.90,0.99:')
-    depends_on('py-texttable@1.4.0:', type='run', when='@:0.90,0.99:')
+    depends_on('py-configparser', type='run', when='@:0.90,0.99: +extras')
+    depends_on('py-graphviz@0.10.1:', type='run', when='@:0.90,0.99: +extras')
+    depends_on('py-matplotlib@3.0.0:', type='run', when='@:0.90,0.99: +extras')
+    depends_on('py-numpy@1.16.0:', type=('build', 'run'), when='@:0.90,0.99: +extras')
+    depends_on('py-onnx@1.3.0:', type='run', when='@:0.90,0.99: +extras')
+    depends_on('py-pandas@0.24.1:', type='run', when='@:0.90,0.99: +extras')
+    depends_on('py-texttable@1.4.0:', type='run', when='@:0.90,0.99: +extras')
     depends_on('py-pytest', type='test', when='@:0.90,0.99:')
-    depends_on('py-protobuf+cpp', type='run', when='@:0.90,0.99:')
+    depends_on('py-protobuf+cpp@3.6.1:', type=('build', 'run'), when='@:0.90,0.99:')
 
     depends_on('py-breathe', type='build', when='+docs')
     depends_on('py-m2r', type='build', when='+docs')
 
     depends_on('cereal')
+    depends_on('catch2')
```

Comment:
This is a test framework right? Is it only needed if you want to run the unit tests? In that case, you could add `type='test'`

Reply:
thanks for the recommendation.